File: 143-reorder-list/reorder-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        if not head:
            return

        arr = []
        p = head
        while p:
            arr.append(p.val)
            p = p.next
        
        p = head
        i = 0
        j = len(arr)-1
        while i<=j:
            if i != j:
                p.val = arr[i]
                p = p.next
            p.val = arr[j]
            p = p.next
            i += 1
            j -= 1
        

File: 143-reorder-list/reorder-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public void reorderList(ListNode head) {
        if(head == null){
            return;
        }

        ArrayList<Integer> arr = new ArrayList<Integer>();
        ListNode p = head;

        while(p != null){
            arr.add(p.val);
            p = p.next;
        }

        p = head;

        int i = 0;
        int j = arr.size()-1;

        while (i<=j){
            if(i != j){
                p.val = arr.get(i);
                p = p.next;
            }
            p.val = arr.get(j);
            p = p.next;
            i++;
            j--;
        }
    }
}